org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/extensions/contentmodel/XMLExternalTest.java

```diff
@@ -0,0 +1,219 @@
+/*******************************************************************************
+* Copyright (c) 2019 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lsp4xml.extensions.contentmodel;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.URI;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import org.eclipse.lsp4j.DidChangeWatchedFilesParams;
+import org.eclipse.lsp4j.DidOpenTextDocumentParams;
+import org.eclipse.lsp4j.FileChangeType;
+import org.eclipse.lsp4j.FileEvent;
+import org.eclipse.lsp4j.MessageActionItem;
+import org.eclipse.lsp4j.MessageParams;
+import org.eclipse.lsp4j.PublishDiagnosticsParams;
+import org.eclipse.lsp4j.ShowMessageRequestParams;
+import org.eclipse.lsp4j.TextDocumentItem;
+import org.eclipse.lsp4j.services.LanguageClient;
+import org.eclipse.lsp4xml.XMLLanguageServer;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.After;
+import org.junit.Test;
+
+/**
+ * This class tests the functionality where an XML file undergoes
+ * validation against its schema when the XSD schema file has been modified
+ * and saved externally through means, not within the LS client.
+ * 
+ * @author David Kwon
+ *
+ */
+public class XMLExternalTest {
+
+	private String tempDirPath = "./src/test/resources/temp";
+	private String xsdPath = tempDirPath + "/sequence.xsd";
+
+	private String xsdContents =
+	"<?xml version=\"1.0\" encoding=\"utf-8\" ?>\n" +
+	"<xs:schema\n" +
+	"    xmlns:xs=\"http://www.w3.org/2001/XMLSchema\"\n" +
+	"    elementFormDefault=\"qualified\">\n" +
+	"  <xs:element name=\"root\">\n" +
+	"    <xs:complexType>\n" +
+	"      <xs:sequence>\n" +
+	"        <xs:element name=\"tag\"/>\n" +
+	"        <xs:element\n" +
+	"            name=\"optional\"\n" +
+	"            minOccurs=\"0\"\n" +
+	"            maxOccurs=\"3\"/>\n" +
+	"      </xs:sequence>\n" +
+	"    </xs:complexType>\n" +
+	"  </xs:element>\n" +
+	"</xs:schema>";
+
+	private String xmlContents = 
+	"<?xml version=\"1.0\" encoding=\"UTF-8\" ?>\n" +
+	"<root xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xsi:noNamespaceSchemaLocation=\"sequence.xsd\">\n" +
+	"  <tag></tag>\n" +
+	"  <optional></optional>\n" +
+	"  <optional></optional>\n" +
+	"  <optional></optional>\n" +
+	"</root>";
+
+	/**
+	 * Creates temporary directory containing the xsd file for
+	 * the test
+	 */
+	@Before
+	public void before() {
+		deleteTempDirIfExists();
+		createTempDir();
+
+		try {
+			createXSDFile();
+		} catch (FileNotFoundException e) {
```

Comment:
no need to catch the exception, just add `throws` to the method's signature, the test will fail anyway